lib/Echidna/Solidity.hs

```diff
@@ -63,27 +64,34 @@ data SolConf = SolConf { _contractAddr :: Addr   -- ^ Contract address to use
                        , _sender       :: [Addr] -- ^ Sender addresses to use
                        , _prefix       :: Text   -- ^ Function name prefix used to denote tests
                        , _solcArgs     :: String -- ^ Args to pass to @solc@
+                       , _quiet        :: Bool   -- ^ Option to suppress some output and stderr of @solc@
                        }
 makeLenses ''SolConf
 
 -- | Given a file, try to compile it and get a list of its contracts, throwing exceptions if necessary.
 contracts :: (MonadIO m, MonadThrow m, MonadReader x m, Has SolConf x) => FilePath -> m [SolcContract]
-contracts fp = view (hasLens . solcArgs) >>= liftIO . solc >>= (\case
-  Nothing -> throwM CompileFailure
-  Just m  -> pure . toList $ fst m) where
-    solc a = readSolc =<< writeSystemTempFile "" =<< readProcess "solc" (usual <> words a) ""
-    usual = ["--combined-json=bin-runtime,bin,srcmap,srcmap-runtime,abi,ast", fp]
-    
+contracts fp = do
+  a <- view (hasLens . solcArgs)
+  stfu <- view (hasLens . quiet)
+  pure (a, stfu) >>= liftIO . solc >>= (\case
```

Comment:
Can we pass the `Handle` straight to `solc`? I think that makes our intent much more clear. It could also let us avoid binding an intermediate variable here.

Reply:
Done. (@japesinator)